src/test/java/seedu/address/storage/JsonAdaptedPropertyTest.java

```diff
@@ -0,0 +1,203 @@
+package seedu.address.storage;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static seedu.address.storage.JsonAdaptedProperty.INCORRECT_CLIENT_FIELD_MESSAGE;
+import static seedu.address.testutil.TypicalClients.CLIENT_EVE;
+import static seedu.address.testutil.TypicalProperties.JURONG;
+import static seedu.address.testutil.TypicalProperties.WOODLANDS_CRESCENT;
+
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.name.Name;
+import seedu.address.model.property.Address;
+import seedu.address.model.property.Deadline;
+import seedu.address.model.property.PostalCode;
+import seedu.address.model.property.Type;
+import seedu.address.model.remark.Remark;
+
+public class JsonAdaptedPropertyTest {
+    @Test
+    public void validNoClientPropertyCreateTest() throws IllegalValueException {
+        JsonAdaptedProperty adaptedProperty = new JsonAdaptedProperty(WOODLANDS_CRESCENT);
+        assertEquals(adaptedProperty.toModelType(), WOODLANDS_CRESCENT);
+        // assertEquals(adaptedProperty, expectedAdaptedProperty);
+    }
+
+    @Test
+    public void validWithClientPropertyCreateTest() throws IllegalValueException {
+        JsonAdaptedProperty adaptedProperty = new JsonAdaptedProperty(JURONG);
+        assertEquals(adaptedProperty.toModelType(), JURONG);
+        // assertEquals(adaptedProperty, expectedAdaptedPropertyWithClient);
+    }
+
+    @Test
+    public void invalidNoNameCreateTest() {
+        JsonAdaptedProperty adaptedPropertyNoName =
+                new JsonAdaptedProperty(null,
+                        WOODLANDS_CRESCENT.getPropertyType().toString(),
+                        WOODLANDS_CRESCENT.getAddress().toString(),
+                        "",
+                        WOODLANDS_CRESCENT.getPostalCode().toString(),
+                        WOODLANDS_CRESCENT.getDeadline().toString(),
+                        null,
+                        WOODLANDS_CRESCENT.getTags().stream().map(JsonAdaptedTag::new).collect(Collectors.toList()),
+                        null);
```

Comment:
Perhaps use `PropertyBuilder` here?

Reply:
that gives nullpointerexception, so must build with constructor.